manifests/init.pp

```diff
@@ -52,9 +55,12 @@
 # [*init_style*]
 #   What style of init system your system uses. Set to 'unmanaged' to disable
 #   managing init system files for the consul service entirely.
+#   This is ignored when install_method == 'docker'
 #
 # [*install_method*]
-#   Valid strings: `package` - install via system package
+#   Valid strings: `auto`    - use docker if available, fallback to url if not
```

Comment:
I'm going to veto this.
I think it adds complexity and surprise.
Just have a default method in the module and make users set it "automatically" themselves with their own puppet code.